sqlglot/transforms.py

```diff
@@ -9,6 +9,27 @@
     from sqlglot.generator import Generator
 
 
+def update_from(expression: exp.Expression) -> exp.Expression:
+    """Transform UPDATE FROM to MERGE INTO"""
+    from_expression = expression.find(exp.From)
+    if isinstance(expression, exp.Update) and from_expression:
+        join = from_expression.find(exp.Join)
+        if join:
+            on = join.args.get("on")
+            table = join.find(exp.Table)
+            this = expression.this
+            then = exp.Update(expressions=expression.expressions)
+            when = exp.When(matched=True, then=then)
+            expression = exp.Merge(this=this, using=table, on=on, expressions=[when])
+        else:
+            table = from_expression.find(exp.Table)
+            expression = exp.Update(
+                this=table, expressions=expression.expressions, where=expression.args["where"]
```

Comment:
>  where=expression.args["where"]

What if there's no `where` arg in the `Update`? Not sure why that is the case, but it's not a required arg so if it's missing this will raise a key error.

Reply:
Is there a better technique to get or return None if missing? 
There are lots of patterns in the code base, I get lost sometimes.

Reply:
I think `where=expression.args.get("where")` would suffice here.